Deeploy/CommonExtensions/CodeTransformationPasses/IntrospectiveCodeTransformation.py

```
@@ -146,60 +156,43 @@ def _extractDynamicExpressions(self,
-        makoGlobalReferences = [
-            node for node in makoExpressions
-            if ((node in operatorRepresentation) and type(operatorRepresentation[node]) == str and (
-                operatorRepresentation[node] in ctxt.globalObjects.keys()))
-        ]
+        # Filter buffers from expressions
+        references = [expr for expr in representedExpressions if ctxt.is_buffer(expr)]
+
+        if unrollStructs:
+
+            def _unrollStructReferences(val: Struct) -> List[str]:
+                assert isinstance(val, Struct)
```

Ruff (0.12.2)

Use of `assert` detected

(S101)